projects/plugins/jetpack/_inc/blogging-prompts.php

```diff
@@ -18,3 +20,180 @@ function jetpack_blogging_prompts_add_meta_data( $keys ) {
 	$keys[] = '_jetpack_blogging_prompt_key';
 	return $keys;
 }
+
+add_filter( 'rest_api_allowed_public_metadata', 'jetpack_blogging_prompts_add_meta_data' );
+
+/**
+ * Sets up a new post as an answer to a blogging prompt.
+ *
+ * Called on `wp_insert_post` hook.
+ *
+ * @param int $post_id ID of post being inserted.
+ * @return void
+ */
+function jetpack_setup_blogging_prompt_response( $post_id ) {
+	// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+	$prompt_id = isset( $_GET['answer_prompt'] ) && absint( $_GET['answer_prompt'] ) ? absint( $_GET['answer_prompt'] ) : false;
+
+	if ( ! jetpack_is_new_post_screen() || ! $prompt_id ) {
+		return;
+	}
+
+	if ( jetpack_is_valid_blogging_prompt( $prompt_id ) ) {
+		update_post_meta( $post_id, '_jetpack_blogging_prompt_key', $prompt_id );
+		wp_add_post_tags( $post_id, 'dailyprompt' );
```

Comment:
The answer should be recorded automatically using the `blogging-prompts` mu-plugin in wpcom (there are hooked functions that handle posts from both Simple and Atomic/Jetpack sites). The `_jetpack_blogging_prompt_key` meta is required for recording the answer post, so it can be tied to a specific prompt id.